src/migration.rs

```diff
@@ -0,0 +1,32 @@
+mod version_0000_initial;
+mod version_0001_uuid;
+
+use serde::Deserialize;
+use serde_yaml::Error;
+
+use crate::filesystem;
+
+#[derive(Deserialize)]
+struct Outline {
+    version: Option<usize>,
+}
+
+pub fn migrate(path: &str) -> Result<(), Error> {
+    let mut yaml = filesystem::open(path);
+
+    loop {
+        let outline: Result<Outline, Error> = serde_yaml::from_str(&yaml);
+        let version = outline.map_or(None, |o| o.version).unwrap_or(0);
```

Comment:
```suggestion
        let version = outline.ok().map_or(0, |o| o.version);
```

Reply:
![image](https://user-images.githubusercontent.com/12772118/87250999-8c40ed00-c4a3-11ea-8be1-600b793bd9e3.png)

Reply:
あぁミスった、suggest 出し直す

Reply:
デフォルト値へ落とし込む処理にただ `None` を使いたくないというお気持ち表明にしかならなくなりました
```suggestion
        let version = outline.map_or(0, |o| o.version.unwrap_or(0));
```

Reply:
それだと `0` が 2 回出てくるので個人的にあまり好きじゃないのです……

Reply:
じゃあ `Outline` の方を `version: u32` にしておいて、
`version` がない場合はデシリアライズのエラーとして扱うのはどうですか
これ以上は Issue 立てるべきレベルなので先に Approve しておきます